plugins/out_es/es_conf_parse.c

```diff
@@ -0,0 +1,319 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2015-2022 The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <string.h>
+
+#include <monkey/mk_core/mk_list.h>
+#include <fluent-bit/tls/flb_tls.h>
+#include <fluent-bit/flb_log.h>
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_slist.h>
+#include <fluent-bit/flb_utils.h>
+#include <fluent-bit/flb_base64.h>
+#include <fluent-bit/flb_output.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_aws_credentials.h>
+
+#include "es_conf_parse.h"
+
+int flb_es_conf_set_cloud_credentials(const char *cloud_auth,
+                                      struct flb_elasticsearch_config *ec)
+{
+    /* extract strings */
+    int items = 0;
+    struct mk_list *toks;
+    struct mk_list *head;
+    struct flb_split_entry *entry;
+
+    if (!cloud_auth) {
+        return 0;
+    }
+
+    toks = flb_utils_split((const char *)cloud_auth, ':', -1);
```

Comment:
Not:

```suggestion
    toks = flb_utils_split((const char *) cloud_auth, ':', -1);
```

because https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md requires to follow https://httpd.apache.org/dev/styleguide.html which requires not put space in case of type cast:

> **The Guidelines**
> ...
> There is no whitespace between a cast and the item modified (e.g., `"(int)j"` and not `"(int) j"`).